### DISCUSSION TITLE
Option to disable powerbar autofill on training mode.

### DISCUSSION THREAD

**Discussion by UrsoManso** [ikemen-engine/Ikemen-GO#1820] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

That's a pretty standard training mode feature that Ikemen's lacks for some reason.

### Describe the solution you'd like

Training menu option to disable/enable powerbar autofill.

**Comment by ghost** [comment]
One thing I would like to say about that, its **extremely** rare for people to disable auto-gain for power in training mode, there has not been any case where someone disabled it in their copy of the fighting game.

So it was within reason to assume that it wasn't necessary, because of how little it was used, but now I can see that it's necessary.

As far as I know, it can be disabled within the training code file that is loaded into training mode, because Ikemen allows you to customize your own game modes as you want.

**Reply by UrsoManso** [reply]
Having the option to toggle off autofill on Ikemen would be even more useful considering someone wants to debug a code related to meter gain without having to change to versus mode. But I don't get how exactly you came up with this data of autofill usage on comercial games, it sounds like a baseless assumption, this feature is in pretty much every fighting game's training mode for a reason.

**Comment by potsmugen** [comment, marked as answer]
To be fair Ikemen already has the best of both worlds, with power only refilling if you're idle.

You can edit Training.zss to act like that I guess. Personally I hesitate to add too many training options because it will clutter the menu and we have to support the smaller Winmugen screenpack. If you add power refill option you'll also have to add life settings (low, medium, full).

I think the important thing is that people _can_ do that in their full games if they want. We don't need to have every single feature by default.
